src/factories/artifactNode.ts

```diff
@@ -12,8 +13,8 @@ type ArtifactNodeFactoryOptions = {
 type ArtifactNodeFactoryRenderOptions = {
   selected?: boolean,
   name?: string,
-  type?: ArtifactType,
-}
+  type: ArtifactType,
+} & RunGraphArtifactTypeAndData
```

Comment:
Since type exists on `RunGraphArtifactTypeAndData` It be little more readable and simpler to do this

```ts
{
  selected?: boolean,
  name?: string,
  artifact: RunGraphArtifactTypeAndData
}
```

Then you can just do `options.artifact.type` everywhere.